src/gui/MainWindow.cpp

```diff
@@ -330,6 +334,87 @@ void MainWindow::display(bool invalidateConfig) {
     emit displayed(WId());
 }
 
+void MainWindow::mousePressEvent (QMouseEvent *event) {
+    QWidget::mousePressEvent(event);
+
+    if (event->button() != Qt::LeftButton) {
+        return;
+    }
+
+    if (scrollingEndTimer.isActive()) {
+        return;
+    }
+
+    auto mousePos = event->localPos();
+    auto imageToCopy = imagePickerDrawer->getImageAtPos(mousePos.x(), mousePos.y());
+
+    if (imageToCopy != nullptr) {
+        copyImage(imageToCopy);
+    }
+}
+
+void MainWindow::mouseMoveEvent(QMouseEvent *event) {
+    QWidget::mouseMoveEvent(event);
+
+    if (scrollingEndTimer.isActive()) {
+        return;
+    }
+
+    auto mousePos = event->localPos();
+    auto imageUnderCursor = imagePickerDrawer->getImageAtPos(mousePos.x(), mousePos.y());
+
+    if (imageUnderCursor != nullptr) {
+        if (imageUnderCursor != imagePickerDrawer->getSelectedImage()) {
+            imagePickerDrawer->drawFrame(imageUnderCursor);
+            repaint();
+        }
+
+        setCursor(Qt::PointingHandCursor);
+    } else {
+        setCursor(Qt::ArrowCursor);
+    }
+}
+
+void MainWindow::wheelEvent(QWheelEvent *event) {
+    QWidget::wheelEvent(event);
+    event->accept();
+        
+    setCursor(Qt::SizeVerCursor);
+    scrollingEndTimer.start((int)scrollEndMilis);
+
+    auto numDegrees = event->angleDelta() / 8;
+
+    xScrollAccumulator += numDegrees.x();
+    yScrollAccumulator += numDegrees.y();
+    
+    bool moved = false;
+
+    if (abs(yScrollAccumulator) >= scrollTreshold) {
+        auto direction = yScrollAccumulator < 0 ? ImagePickerMove::DOWN : ImagePickerMove::UP;
+        moved = this->imagePickerDrawer->move(direction);
+        yScrollAccumulator = 0;
+    }
+
+    if (abs(xScrollAccumulator) >= scrollTreshold) {
+        auto direction = xScrollAccumulator < 0 ? ImagePickerMove::RIGHT : ImagePickerMove::LEFT;
+        if(!moved) {
+            moved = this->imagePickerDrawer->move(direction);
+        }
+        xScrollAccumulator = 0;
+    }
+
+    if (moved) {
+        this->repaint();
+    }
+}
+
+void MainWindow::scrollEnd() {
+    yScrollAccumulator = 0;
+    xScrollAccumulator = 0;
+
+    setCursor(Qt::ArrowCursor);
```

Comment:
this creates an invalid state - this should set the cursor to `Qt::PointingHandCursor` if the cursor is hovering over an image